spec/ruby/core/time/at_spec.rb

```diff
@@ -38,6 +38,16 @@
         Time.at(BigDecimal('1.1')).to_f.should == 1.1
       end
     end
+
+    describe "passed Rational" do
+      it "returns Time with correct nanosecond and microsecond" do
+        t1 = Time.at(Rational(1_486_570_508_539_759, 1_000_000))
+        t2 = Time.at(1_486_570_508, 539_759)
+        
+        t1.usec.should == t2.usec
```

Comment:
I think we should just test `t1.nsec.should == 539_759_000` here, and we should test nanoseconds too so `t1.nsec.should == 539_759_123`.
Otherwise maybe both ways to build a Time are incorrect in the same way and the spec would pass.